pandas/core/indexers.py

```diff
@@ -245,6 +245,12 @@ def length_of_indexer(indexer, target=None) -> int:
             step = -step
         return (stop - start + step - 1) // step
     elif isinstance(indexer, (ABCSeries, ABCIndexClass, np.ndarray, list)):
```

Comment:
make this in to a separate elif clause (for list)

Reply:
sure

Reply:
actually, this seems awkward since we end up checking for bool dtype in two palces